spec/acceptance/ios_interface_spec.rb

```diff
@@ -0,0 +1,167 @@
+  end
+
+  it 'Update Two Instances' do
+    mac = if ['2960', '4507', '6503'].include?(device_model)
+            ['', '', '', '']
+          else
+            ['mac_notification_added => true,', 'mac_notification_removed => true,',
+             'mac_notification_added => false,', 'mac_notification_removed => false,']
+          end
+    pp = <<-EOS
+    ios_interface { '#{target}/1':
+      #{mac[0]}
+      #{mac[1]}
+      link_status => false,
+      logging_event => ['trunk-status'],
+      logging_event_link_status => true,
+      ip_dhcp_snooping_trust => false,
+      ip_dhcp_snooping_limit => 1500,
+      flowcontrol_receive => 'desired',
+    }
+    ios_interface { '#{target}/2':
+      #{mac[2]}
+      #{mac[3]}
+      link_status => true,
+      logging_event => ['spanning-tree','subif-link-status'],
+      logging_event_link_status => false,
+      ip_dhcp_snooping_trust => true,
+      ip_dhcp_snooping_limit => false,
+      flowcontrol_receive => 'on',
+    }
+    EOS
+    make_site_pp(pp)
+    run_device(allow_changes: true)
+    # Are we idempotent
+    run_device(allow_changes: false)
+    # Check puppet resource
+    result = run_resource('ios_interface', "#{target}/1")
+    expect(result).to match(%r{mac_notification_added => true}) if result =~ %r{mac_notification_added =>}
+    expect(result).to match(%r{mac_notification_removed => true}) if result =~ %r{mac_notification_removed =>}
+    expect(result).to match(%r{link_status => false})
+    expect(result).to match(%r{logging_event => \['trunk-status'\]})
+    expect(result).to match(%r{logging_event_link_status => true})
+    expect(result).to match(%r{ip_dhcp_snooping_trust => false})
+    expect(result).to match(%r{ip_dhcp_snooping_limit => 1500,})
+    expect(result).to match(%r{flowcontrol_receive => 'desired'}) if result =~ %r{flowcontrol_receive =>}
+    result = run_resource('ios_interface', "#{target}/2")
+    expect(result).to match(%r{mac_notification_added => false}) if result =~ %r{mac_notification_added =>}
+    expect(result).to match(%r{mac_notification_removed => false}) if result =~ %r{mac_notification_removed =>}
+    expect(result).to match(%r{link_status => true})
+    expect(result).to match(%r{logging_event => \['spanning-tree', 'subif-link-status'\]})
+    expect(result).to match(%r{logging_event_link_status => false})
+    expect(result).to match(%r{ip_dhcp_snooping_trust => true})
+    expect(result).to match(%r{ip_dhcp_snooping_limit => false})
+    expect(result).to match(%r{flowcontrol_receive => 'on'}) if result =~ %r{flowcontrol_receive =>}
+  end
+
+  it 'Unset Two Instances' do
+    mac = if ['2960', '4507', '6503'].include?(device_model)
+            ['', '']
+          else
+            ['mac_notification_added => false,', 'mac_notification_removed => false,']
+          end
+    pp = <<-EOS
+    ios_interface { '#{target}/1':
+      #{mac[0]}
+      #{mac[1]}
+      link_status => false,
+      logging_event => 'unset',
```

Comment:
These tests do not have an idempotency check, which I am guessing it is for `logging_event`. In the `instances_from_cli` and the `clean_logging_event` you could have it set as `unset` to ensure it remains idempotent

Reply:
changed